validator_client/src/main.rs

```diff
@@ -62,6 +71,20 @@ fn main() {
         }
     }
 
+    // TODO: Permit loading a custom spec from file.
+    // Custom spec
+    if let Some(spec_str) = matches.value_of("spec") {
+        match spec_str {
+            "foundation" => config.spec = ChainSpec::foundation(),
+            "few_validators" => config.spec = ChainSpec::few_validators(),
+            // Should be impossible
```

Comment:
This looks like a good opportunity for the [`unreachable!()`](https://doc.rust-lang.org/std/macro.unreachable.html) macro.

```rust
if let Some(spec_str) = matches.value_of("spec") {
        match spec_str {
            "foundation" => config.spec = ChainSpec::foundation(),
            "few_validators" => config.spec = ChainSpec::few_validators(),
            // Should be impossible due to clap's `possible_values(..)` function.
            _ => unreachable!()
        };
}
```

Reply:
yep :+1: